libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogManager.cs

```diff
@@ -0,0 +1,109 @@
+        private readonly ResourceExplorer _resourceExplorer;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogManager"/> class.
+        /// </summary>
+        /// <param name="adaptiveDialogId">The id of the <see cref="AdaptiveDialog"/> to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="languageGeneratorId">The id of the <see cref="LanguageGenerator"/> to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="conversationState">A <see cref="ConversationState"/> implementation.</param>
+        /// <param name="userState">A <see cref="UserState"/> implementation.</param>
+        /// <param name="skillConversationIdFactoryBase">A <see cref="SkillConversationIdFactoryBase"/> implementation.</param>
+        /// <param name="botFrameworkAuthentication">A <see cref="BotFrameworkAuthentication"/> used to obtain a client for making calls to Bot Builder Skills.</param>
+        /// <param name="scopes">Custom <see cref="MemoryScope"/> implementations that extend the memory system.</param>
+        /// <param name="pathResolvers">Custom <see cref="IPathResolver"/> that add new resolvers path shortcuts to memory scopes.</param>
+        /// <param name="dialogs">Custom <see cref="Dialog"/> that will be added to the root DialogSet.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        public AdaptiveDialogManager(
+            string adaptiveDialogId,
+            string languageGeneratorId,
+            string defaultLocale,
+            ResourceExplorer resourceExplorer,
+            ConversationState conversationState,
+            UserState userState,
+            SkillConversationIdFactoryBase skillConversationIdFactoryBase,
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            IEnumerable<MemoryScope> scopes = default,
+            IEnumerable<IPathResolver> pathResolvers = default,
+            IEnumerable<Dialog> dialogs = default,
+            ILogger logger = null)
+        {
+            this._defaultLocale = defaultLocale ?? "en";
+            this._adaptiveDialogId = adaptiveDialogId ?? throw new ArgumentNullException(nameof(adaptiveDialogId));
+            this._botFrameworkAuthentication = botFrameworkAuthentication ?? throw new ArgumentNullException(nameof(botFrameworkAuthentication));
+            this._resourceExplorer = resourceExplorer ?? throw new ArgumentNullException(nameof(resourceExplorer));
+            this._logger = logger ?? NullLogger<AdaptiveDialogManager>.Instance;
+
+            this.InitialTurnState.Add(this._resourceExplorer);
```

Comment:
What is the difference between InitialTurnState and TurnState? Who is supposed to have access to InitialTurnState and who doesn't? 
There seem to be differences between what's being set in InitialTurnState here and what the current AdaptiveDialogBot initializes in SetupTurnState (which is attempting to make explicit what any adaptive dialog should count on in order to work).

Reply:
Either way works.  InitialTurnState is exactly what it says, it is properties that are promoted into each turn, just like SetupTurnState() does.